db/seeds.rb

```diff
@@ -16,11 +16,75 @@
 end
 
 # Prioritize the following models (in this order) such that models depending on them get created after
-priority = [PostType, CloseReason, License, TagSet, PostHistoryType, User, Filter]
+priority = [PostType, CloseReason, License, TagSet, PostHistoryType, User, Ability, CommunityUser, Filter]
 sorted = files.zip(types).to_h.sort do |a, b|
   (priority.index(a.second) || 999) <=> (priority.index(b.second) || 999)
 end.to_h
 
+def expand_communities(type, seed)
+  if type.column_names.include?('community_id') && !seed.include?('community_id')
+    # if model includes a community_id, create the seed for every community
+    Community.all.map { |c| seed.deep_symbolize_keys.merge(community_id: c.id) }
+  else
+    # otherwise, no need to worry, just create it
+    [seed]
+  end
+end
+
+def expand_ids(type, seeds)
+  # Transform all _id relations into the actual rails objects to pass validations
+  seeds.map do |seed|
+    columns = type.column_names.select { |name| name.match(/^.*_id$/) }
+    new_seed = seed.deep_symbolize_keys
+    columns.each do |column|
+      begin
+        column_type_name = column.chomp('_id')
+        column_type = column_type_name.classify.constantize
+        new_seed = new_seed.except(column.to_sym)
+                           .merge(column_type_name.to_sym => column_type.unscoped.find(seed[column.to_sym]))
+      rescue StandardError
+        # Either the type does not exist or the value specified as the id is not valid, ignore.
+        next
+      end
+    end
+    new_seed
+  end
+end
+
+def create_objects(type, seed)
+  seeds = expand_communities(type, seed)
+  seeds = expand_ids(type, seeds)
+
+  # Actually create the objects and count successes
+  objs = type.create seeds
+
+  skipped = objs.select { |o| o.errors.any? }.size
+  created = objs.select { |o| !o.errors.any? }.size
+
+  [created, skipped]
+end
+
+def ensure_system_user_abilities
+  system_users = CommunityUser.unscoped.where(user_id: -1)
+
+  system_users.each do |su|
+    abilities = Ability.unscoped
+      .where(internal_id: ['everyone', 'mod', 'unrestricted'])
+      .where(community_id: su.community_id)
+
+    user_abilities = UserAbility.unscoped.where(community_user_id: su.id)
+
+    abilities.each do |ab|
+      unless user_abilities.any? { |ua| ua.ability_id == ab.id }
```

Comment:
I'm having trouble understanding this line of code.  Dropping this marker here so I can ask you about it in chat.  (I'm assuming this is a "me" problem and not a code clarity problem.)

Reply:
Perfect, thanks! I'll change it to `nil` then - level 4 was just a placeholder value until someone with production access could verify.

Reply:
This part skips any user ability that is already granted - there might be a more elegant way of doing this, but I couldn't think of anything better (yet)